File: two-sum/java/two-sum.java
public int[] twoSum(int[] nums, int target) {
    HashMap<Integer, Integer> numbersMap = new HashMap<>();
    for(int i = 0; i < nums.length; i++){
        numbersMap.put(nums[i], i);
    }
    for(int i = 0; i < nums.length; i++){
        if(numbersMap.containsKey(target - nums[i]) && numbersMap.get(target - nums[i]) != i){
            return new int[] {i, numbersMap.get(target - nums[i])};
        }
    }
    return null;
}

File: two-sum/python/two-sum.py
def twoSum(self, nums, target):
    """
    :type nums: List[int]
    :type target: int
    :rtype: List[int]
    """
    numbersMap = {}
    for i in range(len(nums)):
        numbersMap[nums[i]] = i
        
    for i in range(len(nums)):
        complement = target - nums[i]
        if complement in numbersMap and numbersMap[complement] != i:
            return [i, numbersMap[complement]]
        
    return []